lib/tasks/upgrade.rake

```diff
@@ -944,12 +1063,161 @@ namespace :upgrade do
+  task migrate_plan_org_and_funder: :environment do
+    plans = Plan.includes(template: :org, roles: :user)
+                .joins(template: :org, roles: :user)
+
+    p "Attaching Plans to Orgs ... this can take in excess of 5 minutes depending on how many plans you have."
+    Parallel.map(plans, in_threads: 8) do |plan|
+      next if plan.org_id.present?
+
+      # Parallel has trouble with ActiveRecord lazy loading
+      require "plan" unless Object.const_defined?("Plan")
+      require "role" unless Object.const_defined?("Role")
+      require "perm" unless Object.const_defined?("Perm")
+      require "user" unless Object.const_defined?("User")
+      @reconnected ||= Plan.connection.reconnect! || true
+
+      next unless plan.owner.present? && plan.owner.org.present?
+
+      plan.update(org_id: plan.owner.org.id)
+    end
+
+    p "Attaching Plans to Funders"
+    Parallel.map(plans, in_threads: 8) do |plan|
+      next if plan.funder_id.present?
+
+      # Parallel has trouble with ActiveRecord lazy loading
+      require "plan" unless Object.const_defined?("Plan")
+      require "template" unless Object.const_defined?("Template")
+      require "org" unless Object.const_defined?("Org")
+      @reconnected ||= Plan.connection.reconnect! || true
+
+      next unless plan.funder_name.present? || plan.template.org.funder?
+
+      funder_id = plan.template.org.id if plan.template.org.funder?
+
+      if plan.funder_name.present? && !funder_id.present?
+        matches = OrgSelection::SearchService.search_externally(search_term: plan.funder_name)
+        # Only allow results that INCLUDE the search term in parenthesis
+        matches = matches.select do |result|
+          result[:weight] <= 1 && result[:name].include?("(#{plan.funder_name})")
+        end
+
+        org = OrgSelection::HashToOrgService.to_org(hash: matches.first, allow_create: true) if matches.any?
+        org = create_org(org, matches.first)  if org.present? && org.valid?
+        funder_id = org.id if org.present?
+      end
+
+      plan.update(funder_id: funder_id) if funder_id.present?
     end
+    p "Complete"
+  end
+
+  desc "Migrate the Plans grant_number to an Identifier"
+  task migrate_plan_grants: :environment do
+    plans = Plan.where.not(grant_number: nil).where.not(grant_number: "")
+
+    p "Converting Plan.grant_number into Identifiers"
+    #Parallel.map(plans, in_threads: 8) do |plan|
+    plans.each do |plan|
+      # Parallel has trouble with ActiveRecord lazy loading
+      require "plan" unless Object.const_defined?("Plan")
+      @reconnected ||= Plan.connection.reconnect! || true
+
+      identifier = Identifier.find_or_create_by(
+        identifier_scheme_id: nil, identifiable: plan, value: plan.grant_number
+      )
+      plan.update(grant_id: identifier.id)
+    end
+    p "Complete"
+  end
+
+  desc "Generate stats for all of the 2.2.0 upgrade scripts"
+  task results_2_2_0_part1: :environment do
```

Comment:
These as well.  This will make the upgrade scripts a lot easier to test/follow along with